samza-core/src/main/java/org/apache/samza/container/RunLoopFactory.java

```diff
@@ -52,18 +50,6 @@ public static Runnable createRunLoop(scala.collection.immutable.Map<TaskName, Ta
 
     log.info("Got commit milliseconds: {}.", taskCommitMs);
 
-    int asyncTaskCount = taskInstances.values().count(new AbstractFunction1<TaskInstance, Object>() {
-      @Override
-      public Boolean apply(TaskInstance t) {
-        return t.isAsyncTask();
-      }
-    });
-
-    // asyncTaskCount should be either 0 or the number of all taskInstances
-    if (asyncTaskCount > 0 && asyncTaskCount < taskInstances.size()) {
-      throw new SamzaException("Mixing StreamTask and AsyncStreamTask is not supported");
-    }
```

Comment:
Was this validation moved somewhere else? Or is it no longer necessary?

Reply:
Thanks for asking. I think I should update the PR description to include details on this.

It is no longer necessary. In `SamzaContainer`, we will always wrap instances of `StreamTask` into `AsyncStreamTaskAdapter`. Thus, isAsyncTask is redundant since it is always true.

Reply:
Updated PR description with these details.